tests/conftest.py

```diff
@@ -168,18 +171,44 @@ def hang_url(local_server: LocalHttpServer):
         local_server.hang_forever_stop()
 
 
-@pytest.fixture
-def bad_ssl_url():
-    """
-    Return a URL with an expired SSL certificate.
+@pytest.fixture(scope="session")
+def ssl_context_err_cert_authority_invalid():
+    """Return a SSL context with an invalid certificate authority."""
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+
+    cert_file_name = Path(__file__).parent / "tools" / "cert.pem"
+    key_file_name = Path(__file__).parent / "tools" / "key.pem"
+
+    context.load_cert_chain(cert_file_name, key_file_name)
+
+    return context
+
 
+@pytest.fixture(scope="session")
+def ssl_httpserver_err_cert_authority_invalid(
+        ssl_context_err_cert_authority_invalid):
+    """Return a HTTPServer instance with SSL enabled.
```

Comment:
We should probably change the name of the `HTTPServer` to `HttpServer` to be more inline with the code conventions we have.
```suggestion
    """Return a HttpsServer instance with invalid SSL enabled.
```

Reply:
I think this is non actionable as its lib import.